### ISSUE TITLE
`envsubst` is not installed on OSX by default, causing command not found when following the book

### ISSUE THREAD

**Issue by b2jrock** [kubernetes-sigs/cluster-api#1827]
**What steps did you take and what happened:**
When following the installation in the book, under "Install infrastructure provider" under multiple providers, envsubst is used to fill in some values in the yaml. Since this isn't installed by default on osx, I get a "command not found".

**What did you expect to happen:**
Not getting a "command not found".

**Anything else you would like to add:**
This is just some documentation clarification. Maybe a prerequisites section would be helpful. This is easily worked around with homebrew:
```bash
brew install gettext
brew link --force gettext
```

**Environment:**

- Cluster-api version: 
- Minikube/KIND version: 0.6.0
- Kubernetes version: (use `kubectl version`): 1.16.3
- OS (e.g. from `/etc/os-release`): MacOS 10.14.6

/kind bug

**Comment by prankul88** [comment]
@b2jrock Probably best to add it in Prerequisites section.

/assign

**Comment by ncdc** [comment]
Not sure if this is helpful, but for CAPA's `examples/generate.sh` script, we work around this by defining a function in the script to supply the functionality:

```
envsubst() {
  python -c 'import os,sys;[sys.stdout.write(os.path.expandvars(l)) for l in sys.stdin]'
}
```

And then it's used the same way as a standalone binary:

```
kustomize build "${SOURCE_DIR}/cluster" | envsubst > "${CLUSTER_GENERATED_FILE}"
```

**Comment by detiber** [comment]
There was recently a PR that added a blurb to the installation guide as well stating that `envsubst` is provided by the `gettext` package and available via Brew on MacOS here: https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/tasks/installation.md#create-the-components

It isn't in the published version of the docs, though.

**Comment by b2jrock** [comment]
Looks like that PR was merged in #1815 right around when I filed this one. Safe to close, I think.
